src/test/mir-opt/matches_reduce_branches.foo.MatchBranchSimplification.diff

```diff
@@ -0,0 +1,64 @@
+- // MIR for `foo` before MatchBranchSimplification
++ // MIR for `foo` after MatchBranchSimplification
+  
+  fn foo(_1: std::option::Option<()>) -> () {
+      debug bar => _1;                     // in scope 0 at $DIR/matches_reduce_branches.rs:3:8: 3:11
+      let mut _0: ();                      // return place in scope 0 at $DIR/matches_reduce_branches.rs:3:25: 3:25
+      let mut _2: bool;                    // in scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
+      let mut _3: isize;                   // in scope 0 at $DIR/matches_reduce_branches.rs:4:22: 4:26
+  
+      bb0: {
+          StorageLive(_2);                 // scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
+          _3 = discriminant(_1);           // scope 0 at $DIR/matches_reduce_branches.rs:4:22: 4:26
+-         switchInt(move _3) -> [0_isize: bb2, otherwise: bb1]; // scope 0 at $DIR/matches_reduce_branches.rs:4:22: 4:26
++         goto -> bb2;                     // scope 0 at $DIR/matches_reduce_branches.rs:4:22: 4:26
+      }
+  
+      bb1: {
+          _2 = const false;                // scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
+                                           // ty::Const
+                                           // + ty: bool
+                                           // + val: Value(Scalar(0x00))
+                                           // mir::Constant
+                                           // + span: $SRC_DIR/core/src/macros/mod.rs:LL:COL
+                                           // + literal: Const { ty: bool, val: Value(Scalar(0x00)) }
+          goto -> bb3;                     // scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
+      }
+  
+      bb2: {
+-         _2 = const true;                 // scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
++         _2 = Eq(move _3, const 0_isize); // scope 0 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
+                                           // ty::Const
+-                                          // + ty: bool
+-                                          // + val: Value(Scalar(0x01))
++                                          // + ty: isize
```

Comment:
due to this isize constant here, you need https://github.com/rust-lang/rust/tree/41094ea9ff2c72d6c5d31566b50b046d0f96fce7/src/test/mir-opt#--blessable-test-format to generate one file for 32 bit and one for 64 bit. Run the tests with `--target i686-unknown-linux-gnu` to generate the 32 bit file